### DISCUSSION TITLE
Mac App Store builds in TestFlight

### DISCUSSION THREAD

**Discussion by kasparv** [airsdk/Adobe-Runtime-Support#2218] [Q&A] [answered]
Has anyone looked into how to make AIR-built Mac App Store apps available for testing in Mac TestFlight?
Are there any specific requirements for that?
For now, when I upload Mac builds to App Store Connect, their TestFlight status is "Not Available for Testing" although they can be made available through Mac App Store...

**Comment by kasparv** [comment, marked as answer]
I looked into how other non-Xcode solutions do that and found out that it was just about having the right entitlements.
After adding `com.apple.security.application-groups`, `com.apple.application-identifier` and `com.apple.developer.team-identifier keys` the AIR app became available for Mac TestFlight.
For anyone looking for the same answer - here's a basic root entitlements example:

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>com.apple.security.app-sandbox</key>
	<true/>
    <key>com.apple.security.application-groups</key>
    <string>TEAMID.com.app.appname</string>
    <key>com.apple.application-identifier</key>
    <string>TEAMID.com.app.appname</string>
    <key>com.apple.developer.team-identifier</key>
    <string>TEAMID</string>
	<key>com.apple.security.network.client</key>
	<true/>
	<key>com.apple.security.cs.allow-jit</key>
	<true/>
	<key>com.apple.security.cs.allow-unsigned-executable-memory</key>
	<true/>
	<key>com.apple.security.cs.disable-library-validation</key>
	<true/>
</dict>
</plist>
```

**Reply by marchbold** [reply]
Came here to say that, however I don't think you need the application-groups one though unless you are using app groups in your app. 

The key ones are 

```xml
<key>com.apple.security.app-sandbox</key>
<true/>
<key>com.apple.application-identifier</key>
<string>TEAMID.com.app.appname</string>
<key>com.apple.developer.team-identifier</key>
<string>TEAMID</string>
```

These ones are important to make sure the AIR app works in the macOS sandbox. 

```xml
<key>com.apple.security.network.client</key>
<true/>
<key>com.apple.security.cs.allow-jit</key>
<true/>
<key>com.apple.security.cs.allow-unsigned-executable-memory</key>
<true/>
<key>com.apple.security.cs.disable-library-validation</key>
<true/>
```

**Comment by kasparv** [comment]
I was able to make the Mac build available for App Store Connect users in TestFlight but for external testers the app must be submitted for review and that currently is not possible since the captive AIR runtime is built with an outdated version of Xcode
```
	<key>DTXcode</key>
	<string>1300</string>
	<key>DTXcodeBuild</key>
	<string>13A233</string>
```

I have also created a bug issue https://github.com/airsdk/Adobe-Runtime-Support/issues/2258
